# COMP30760, Assignment 1, Task 1
# by Muhammad Khan, 21338981
In this assignment, I will be collecting and analysing historical data on cryptocurrencies dating back to 2011. The API I have chosen is the coinlayer one, https://coinlayer.com/. The API only has data from 2011 onwards. In this notebook I will be covering the Data Collection aspect of the assignment.

In [81]:
import urllib.request
import json
import pprint
import pandas as pd 
import csv

Now I will request data from the API from various years.

# 2010

In [82]:
#requesting data for 20th April, 2010
Date1 = '2010-04-20'
access_key = '94a47b4780c2538d0a2af7c4286ee273';
url = 'http://api.coinlayer.com/api/' + Date1 + '?access_key=' + access_key
response1 = urllib.request.urlopen(url)

In [83]:
j1 = response1.read().decode('utf-8')
j_a = json.loads(j1)
j_a

{'success': False,
 'error': {'code': 202, 'type': 'failed_getting_crypto_currency_data'}}

As we can see here, the API does not have historical data from before 2011. I picked April 20th at random, and kept it consistent throughout the assignment.

In [84]:
#data in a dataframe
df1 = pd.DataFrame.from_dict(j_a)
df1


success                                error
code    False                                  202
type    False  failed_getting_crypto_currency_data

# 2011

In [85]:
#requesting data for 20th April, 2011.
Date2 = '2011-04-20'
url = 'http://api.coinlayer.com/api/' + Date2 + '?access_key=' + access_key
response2 = urllib.request.urlopen(url)

In [86]:
j2 = response2.read().decode('utf-8')
j_b = json.loads(j2)
j_b

{'success': True,
 'terms': 'https://coinlayer.com/terms',
 'privacy': 'https://coinlayer.com/privacy',
 'timestamp': 1303257600,
 'target': 'USD',
 'historical': True,
 'date': '2011-04-20',
 'rates': {'BTC': 1.14}}

As the API covers historical data from 2011, we can now see BTC(bitcoin) here.For some context, it was the first cryptocurrency and made in 2009. I have picked a few years spaced out a few years apart to analyse them.

In [87]:
#data in a dataframe
df2 = pd.DataFrame.from_dict(j_b)
df2

success                        terms                        privacy  \
BTC     True  https://coinlayer.com/terms  https://coinlayer.com/privacy   

      timestamp target  historical        date  rates  
BTC  1303257600    USD        True  2011-04-20   1.14

Here I converted the data for 2011 into a csv file:

In [122]:
#name of file
csv_filename = '2011.csv'

#converting df to csv
df2.to_csv(csv_filename, index_label='Currency')

print(f'DataFrame has been saved as "{csv_filename}"')

DataFrame has been saved as "2011.csv"


# 2012

In [93]:
#requesting data for 20th April, 2012
Date3 = '2012-04-20'
url = 'http://api.coinlayer.com/api/' + Date3 + '?access_key=' + access_key
response3 = urllib.request.urlopen(url)

In [94]:
j3 = response3.read().decode('utf-8')
j_c = json.loads(j3)
j_c

{'success': True,
 'terms': 'https://coinlayer.com/terms',
 'privacy': 'https://coinlayer.com/privacy',
 'timestamp': 1334880000,
 'target': 'USD',
 'historical': True,
 'date': '2012-04-20',
 'rates': {'BTC': 5.35}}

Here we can see that the price of bitcoin has gone up to $5.35

In [95]:
#data in a dataframe
df3 = pd.DataFrame.from_dict(j_c)
df3


success                        terms                        privacy  \
BTC     True  https://coinlayer.com/terms  https://coinlayer.com/privacy   

      timestamp target  historical        date  rates  
BTC  1334880000    USD        True  2012-04-20   5.35

Here I converted the data for 2012 into a csv file:

In [121]:
#name of file
csv_filename = '2012.csv'

#converting df to csv
df3.to_csv(csv_filename, index_label='Currency') 

print(f'DataFrame has been saved as "{csv_filename}"')

DataFrame has been saved as "2012.csv"


# 2013

In [97]:
#requesting data for 20th April, 2013
Date4 = '2013-04-20'
url = 'http://api.coinlayer.com/api/' + Date4 + '?access_key=' + access_key
response4 = urllib.request.urlopen(url)

In [98]:
j4 = response4.read().decode('utf-8')
j_d = json.loads(j4)
j_d

{'success': True,
 'terms': 'https://coinlayer.com/terms',
 'privacy': 'https://coinlayer.com/privacy',
 'timestamp': 1366416000,
 'target': 'USD',
 'historical': True,
 'date': '2013-04-20',
 'rates': {'BTC': 126.62}}

Again, no more cryptocurrency coins added, but the price of bitcoin is increasing. 

In [99]:
#data in a dataframe
df4 = pd.DataFrame.from_dict(j_d)
df4


success                        terms                        privacy  \
BTC     True  https://coinlayer.com/terms  https://coinlayer.com/privacy   

      timestamp target  historical        date   rates  
BTC  1366416000    USD        True  2013-04-20  126.62

Here I converted the data for 2013 into a csv file:

In [120]:
#name of file
csv_filename = '2013.csv'

#converting df to csv
df4.to_csv(csv_filename, index_label='Currency')

print(f'DataFrame has been saved as "{csv_filename}"')

DataFrame has been saved as "2013.csv"


# 2016

In [101]:
#requesting data for 20th April, 2016
Date5 = '2016-04-20'
url = 'http://api.coinlayer.com/api/' + Date5 + '?access_key=' + access_key
response5 = urllib.request.urlopen(url)

In [102]:
j5 = response5.read().decode('utf-8')
j_e = json.loads(j5)
j_e

{'success': True,
 'terms': 'https://coinlayer.com/terms',
 'privacy': 'https://coinlayer.com/privacy',
 'timestamp': 1461110400,
 'target': 'USD',
 'historical': True,
 'date': '2016-04-20',
 'rates': {'BTC': 441.16,
  'DASH': 6.2,
  'DOGE': 0.000225,
  'ETH': 8.54,
  'LTC': 3.27,
  'STEEM': 0.8325,
  'VEN': 0.09091,
  'XMR': 1.06,
  'XRP': 0.006951,
  'XVG': 0.00012}}

In [103]:
#data in a dataframe
df5 = pd.DataFrame.from_dict(j_e)
df5


success                        terms                        privacy  \
BTC       True  https://coinlayer.com/terms  https://coinlayer.com/privacy   
DASH      True  https://coinlayer.com/terms  https://coinlayer.com/privacy   
DOGE      True  https://coinlayer.com/terms  https://coinlayer.com/privacy   
ETH       True  https://coinlayer.com/terms  https://coinlayer.com/privacy   
LTC       True  https://coinlayer.com/terms  https://coinlayer.com/privacy   
STEEM     True  https://coinlayer.com/terms  https://coinlayer.com/privacy   
VEN       True  https://coinlayer.com/terms  https://coinlayer.com/privacy   
XMR       True  https://coinlayer.com/terms  https://coinlayer.com/privacy   
XRP       True  https://coinlayer.com/terms  https://coinlayer.com/privacy   
XVG       True  https://coinlayer.com/terms  https://coinlayer.com/privacy   

        timestamp target  historical        date       rates  
BTC    1461110400    USD        True  2016-04-20  441.160000  
DASH   1461110400    USD        True  2016-04-20    6.200000  
DOGE   1461110400    USD        True  2016-04-20    0.000225  
ETH    1461110400    USD        True  2016-04-20    8.540000  
LTC    1461110400    USD        True  2016-04-20    3.270000  
STEEM  1461110400    USD        True  2016-04-20    0.832500  
VEN    1461110400    USD        True  2016-04-20    0.090910  
XMR    1461110400    USD        True  2016-04-20    1.060000  
XRP    1461110400    USD        True  2016-04-20    0.006951  
XVG    1461110400    USD        True  2016-04-20    0.000120

As the number of cryptocurrencies increase, I decided to portray their rates a bit more clearly(just for my own intuitive purposes) by iterating through j_e:

In [24]:
for currency, rate in j_e['rates'].items():
    print(f"{currency} : {rate}")


BTC : 441.16
DASH : 6.2
DOGE : 0.000225
ETH : 8.54
LTC : 3.27
STEEM : 0.8325
VEN : 0.09091
XMR : 1.06
XRP : 0.006951
XVG : 0.00012


Here I converted the data for 2016 into a csv file

In [119]:
#name of file
csv_filename = '2016.csv'

#converting df to csv
df5.to_csv(csv_filename, index_label='Currency') 

print(f'DataFrame has been saved as "{csv_filename}"')

DataFrame has been saved as "2016.csv"


# 2018

In [25]:
#requesting data for 20th April, 2018
Date6 = '2018-04-20'
url = 'http://api.coinlayer.com/api/' + Date6 + '?access_key=' + access_key
response6 = urllib.request.urlopen(url)


In [26]:
j6 = response6.read().decode('utf-8')
j_f = json.loads(j6)
j_f


{'success': True,
 'terms': 'https://coinlayer.com/terms',
 'privacy': 'https://coinlayer.com/privacy',
 'timestamp': 1524268743,
 'target': 'USD',
 'historical': True,
 'date': '2018-04-20',
 'rates': {'ABC': 70,
  'ACP': 0.014931,
  'ACT': 0.27998,
  'ACT*': 0.276836,
  'ADA': 0.304571,
  'ADCN': 0.003209,
  'ADL': 243,
  'ADX': 0.912246,
  'ADZ': 0.03386,
  'AE': 2.070437,
  'AGI': 0.20719,
  'AIB': 0.005626,
  'AIDOC': 0.049116,
  'AION': 3.354242,
  'AIR': 0.00936,
  'ALT': 51.99,
  'AMB': 0.523832,
  'AMM': 0.260013,
  'ANT': 4.2,
  'APC': 0.003394,
  'APPC': 0.536898,
  'ARC': 0.050711,
  'ARCT': 0.009254,
  'ARDR': 0.399527,
  'ARK': 3.152324,
  'ARN': 1.482354,
  'ASAFE2': 0.0133,
  'AST': 0.509543,
  'ATB': 0.159,
  'ATM': 0.0045,
  'AURS': 0.352867,
  'AVT': 1.3899,
  'BAR': 0.00563,
  'BASH': 0.0056,
  'BAT': 0.406462,
  'BAY': 0.103565,
  'BBP': 0.00461,
  'BCD': 4.58478,
  'BCH': 1124.795801,
  'BCN': 0.005135,
  'BCPT': 0.558481,
  'BEE': 0.000103,
  'BIO': 0.00626,
  'B

In [21]:
for currency, rate in j_f['rates'].items():
    print(f"{currency} : {rate}")

ABC : 70
ACP : 0.014931
ACT : 0.27998
ACT* : 0.276836
ADA : 0.304571
ADCN : 0.003209
ADL : 243
ADX : 0.912246
ADZ : 0.03386
AE : 2.070437
AGI : 0.20719
AIB : 0.005626
AIDOC : 0.049116
AION : 3.354242
AIR : 0.00936
ALT : 51.99
AMB : 0.523832
AMM : 0.260013
ANT : 4.2
APC : 0.003394
APPC : 0.536898
ARC : 0.050711
ARCT : 0.009254
ARDR : 0.399527
ARK : 3.152324
ARN : 1.482354
ASAFE2 : 0.0133
AST : 0.509543
ATB : 0.159
ATM : 0.0045
AURS : 0.352867
AVT : 1.3899
BAR : 0.00563
BASH : 0.0056
BAT : 0.406462
BAY : 0.103565
BBP : 0.00461
BCD : 4.58478
BCH : 1124.795801
BCN : 0.005135
BCPT : 0.558481
BEE : 0.000103
BIO : 0.00626
BLC : 0.072132
BLOCK : 19.0872
BLU : 0.00283
BLZ : 0.561234
BMC : 0.62688
BNB : 12.585073
BNT : 3.858638
BOST : 0.012
BQ : 0.003699
BQX : 3.3
BRD : 0.690196
BRIT : 0.15
BT1 : 0
BT2 : 0
BTC : 8860.863277
BTCA : 0.00074
BTCS : 0.182629
BTCZ : 0.00417
BTG : 70.045213
BTLC : 0.13
BTM : 0.951745
BTM* : 0.952201
BTQ : 0.09
BTS : 0.276955
BTX : 9.174786
BURST : 0.029787
CALC : 0.00

In [28]:
#data in a dictionary format
df6 = pd.DataFrame.from_dict(j_f)
df6


success                        terms                        privacy  \
611      True  https://coinlayer.com/terms  https://coinlayer.com/privacy   
ABC      True  https://coinlayer.com/terms  https://coinlayer.com/privacy   
ACP      True  https://coinlayer.com/terms  https://coinlayer.com/privacy   
ACT      True  https://coinlayer.com/terms  https://coinlayer.com/privacy   
ACT*     True  https://coinlayer.com/terms  https://coinlayer.com/privacy   
...       ...                          ...                            ...   
ZEN      True  https://coinlayer.com/terms  https://coinlayer.com/privacy   
ZIL      True  https://coinlayer.com/terms  https://coinlayer.com/privacy   
ZNY      True  https://coinlayer.com/terms  https://coinlayer.com/privacy   
ZRX      True  https://coinlayer.com/terms  https://coinlayer.com/privacy   
ZSC      True  https://coinlayer.com/terms  https://coinlayer.com/privacy   

       timestamp target  historical        date      rates  
611   1524268743    USD        True  2018-04-20   0.382624  
ABC   1524268743    USD        True  2018-04-20  70.000000  
ACP   1524268743    USD        True  2018-04-20   0.014931  
ACT   1524268743    USD        True  2018-04-20   0.279980  
ACT*  1524268743    USD        True  2018-04-20   0.276836  
...          ...    ...         ...         ...        ...  
ZEN   1524268743    USD        True  2018-04-20  40.793602  
ZIL   1524268743    USD        True  2018-04-20   0.081021  
ZNY   1524268743    USD        True  2018-04-20   0.033993  
ZRX   1524268743    USD        True  2018-04-20   0.992248  
ZSC   1524268743    USD        True  2018-04-20   0.058538  

[386 rows x 8 columns]

Here I converted the data for 2018 into a csv file

In [118]:
#name of file
csv_filename = '2018.csv'

#converting df to csv
df6.to_csv(csv_filename, index_label='Currency') 

print(f'DataFrame has been saved as "{csv_filename}"')

DataFrame has been saved as "2018.csv"


# 2021

In [30]:
#requesting data for 20th April, 2021
Date7 = '2021-04-20'
url = 'http://api.coinlayer.com/api/' + Date7 + '?access_key=' + access_key
response7 = urllib.request.urlopen(url)


In [31]:
j7 = response7.read().decode('utf-8')
j_g = json.loads(j7)
j_g

{'success': True,
 'terms': 'https://coinlayer.com/terms',
 'privacy': 'https://coinlayer.com/privacy',
 'timestamp': 1618963147,
 'target': 'USD',
 'historical': True,
 'date': '2021-04-20',
 'rates': {'ABC': 59.99,
  'ACP': 0.014931,
  'ACT': 0.024184,
  'ACT*': 0.017178,
  'ADA': 1.264672,
  'ADCN': 0.00013,
  'ADL': 0.01515,
  'ADX': 1.278893,
  'ADZ': 0.0023,
  'AE': 0.374574,
  'AGI': 0,
  'AIB': 0.005626,
  'AIDOC': 0.003389,
  'AION': 0.32983,
  'AIR': 0.001506,
  'ALT': 0.565615,
  'AMB': 0.07301,
  'AMM': 0.016317,
  'ANT': 8.4146,
  'APC': 0.0017,
  'APPC': 0.204305,
  'ARC': 0.0169,
  'ARCT': 0.00061,
  'ARDR': 0.452257,
  'ARK': 2.034231,
  'ARN': 0.064742,
  'ASAFE2': 0.4,
  'AST': 0.394853,
  'ATB': 0.017,
  'ATM': 0,
  'AURS': 0.352867,
  'AVAX': 27.503,
  'AVT': 0,
  'BAR': 0.00685,
  'BASH': 0.0056,
  'BAT': 1.263671,
  'BAY': 0.0644,
  'BBP': 0.0005,
  'BCD': 2.430112,
  'BCH': 942.241398,
  'BCN': 0.000826,
  'BCPT': 0.009038,
  'BEE': 1e-06,
  'BIO': 0.0008,
  'BLC

In [32]:
for currency, rate in j_g['rates'].items():
    print(f"{currency} : {rate}")

ABC : 59.99
ACP : 0.014931
ACT : 0.024184
ACT* : 0.017178
ADA : 1.264672
ADCN : 0.00013
ADL : 0.01515
ADX : 1.278893
ADZ : 0.0023
AE : 0.374574
AGI : 0
AIB : 0.005626
AIDOC : 0.003389
AION : 0.32983
AIR : 0.001506
ALT : 0.565615
AMB : 0.07301
AMM : 0.016317
ANT : 8.4146
APC : 0.0017
APPC : 0.204305
ARC : 0.0169
ARCT : 0.00061
ARDR : 0.452257
ARK : 2.034231
ARN : 0.064742
ASAFE2 : 0.4
AST : 0.394853
ATB : 0.017
ATM : 0
AURS : 0.352867
AVAX : 27.503
AVT : 0
BAR : 0.00685
BASH : 0.0056
BAT : 1.263671
BAY : 0.0644
BBP : 0.0005
BCD : 2.430112
BCH : 942.241398
BCN : 0.000826
BCPT : 0.009038
BEE : 1e-06
BIO : 0.0008
BLC : 0.072132
BLOCK : 14.2395
BLU : 0.00054
BLZ : 0.387888
BMC : 0.050742
BNB : 587.098128
BNT : 6.652558
BOST : 0.048
BQ : 7.775e-05
BQX : 3.81666
BRD : 0.368897
BRIT : 0.03
BT1 : 0
BT2 : 0
BTC : 56466.846102
BTCA : 0.00036
BTCS : 0.01201
BTCZ : 0.000749
BTG : 93.569334
BTLC : 9
BTM : 0.237899
BTM* : 0.122609
BTQ : 0.01
BTS : 0.114615
BTX : 0.429692
BURST : 0.014123
CALC : 0.000

In [130]:
#data in a dataframe
df7 = pd.DataFrame.from_dict(j_g)
df7


success                        terms                        privacy  \
611      True  https://coinlayer.com/terms  https://coinlayer.com/privacy   
ABC      True  https://coinlayer.com/terms  https://coinlayer.com/privacy   
ACP      True  https://coinlayer.com/terms  https://coinlayer.com/privacy   
ACT      True  https://coinlayer.com/terms  https://coinlayer.com/privacy   
ACT*     True  https://coinlayer.com/terms  https://coinlayer.com/privacy   
...       ...                          ...                            ...   
ZEN      True  https://coinlayer.com/terms  https://coinlayer.com/privacy   
ZIL      True  https://coinlayer.com/terms  https://coinlayer.com/privacy   
ZNY      True  https://coinlayer.com/terms  https://coinlayer.com/privacy   
ZRX      True  https://coinlayer.com/terms  https://coinlayer.com/privacy   
ZSC      True  https://coinlayer.com/terms  https://coinlayer.com/privacy   

       timestamp target  historical        date       rates  
611   1618963147    USD        True  2021-04-20    0.389165  
ABC   1618963147    USD        True  2021-04-20   59.990000  
ACP   1618963147    USD        True  2021-04-20    0.014931  
ACT   1618963147    USD        True  2021-04-20    0.024184  
ACT*  1618963147    USD        True  2021-04-20    0.017178  
...          ...    ...         ...         ...         ...  
ZEN   1618963147    USD        True  2021-04-20  117.783645  
ZIL   1618963147    USD        True  2021-04-20    0.192362  
ZNY   1618963147    USD        True  2021-04-20    0.020000  
ZRX   1618963147    USD        True  2021-04-20    1.711158  
ZSC   1618963147    USD        True  2021-04-20    0.000791  

[387 rows x 8 columns]

Here I converted the data for 2021 into a csv file

In [117]:
#name of file
csv_filename = '2021.csv'

#converting df to csv
df7.to_csv(csv_filename, index_label='Currency') 

print(f'DataFrame has been saved as "{csv_filename}"')

DataFrame has been saved as "2021.csv"


# 2023

In [124]:
#requesting data for 20th April, 2023
Date8 = '2023-04-20'
url = 'http://api.coinlayer.com/api/' + Date8 + '?access_key=' + access_key
response8 = urllib.request.urlopen(url)

In [125]:
j8 = response8.read().decode('utf-8')
j_h = json.loads(j8)
j_h

{'success': True,
 'terms': 'https://coinlayer.com/terms',
 'privacy': 'https://coinlayer.com/privacy',
 'timestamp': 1682035146,
 'target': 'USD',
 'historical': True,
 'date': '2023-04-20',
 'rates': {'ABC': 59.99,
  'ACP': 0.014931,
  'ACT': 0.00107,
  'ACT*': 0.017178,
  'ADA': 0.403125,
  'ADCN': 0.00013,
  'ADL': 0.01515,
  'ADX': 0.18734,
  'ADZ': 0.0023,
  'AE': 0.069169,
  'AGI': 0.4267,
  'AIB': 0.005626,
  'AIDOC': 1.627597959e-05,
  'AION': 0.001526,
  'AIR': 0.00976,
  'ALT': 0.565615,
  'AMB': 0.014214,
  'AMM': 0.003387,
  'ANT': 3.67465,
  'APC': 0.0017,
  'APPC': 0.002,
  'ARC': 0.0169,
  'ARCT': 0.00061,
  'ARDR': 0.087504,
  'ARK': 0.81952,
  'ARN': 0.001283,
  'ASAFE2': 0.4,
  'AST': 0.1173,
  'ATB': 0.017,
  'ATM': 3.43895,
  'AURS': 0.352867,
  'AVAX': 18.36,
  'AVT': 1.14,
  'BAR': 4.215,
  'BASH': 0.0056,
  'BAT': 0.25644,
  'BAY': 0.0644,
  'BBP': 0.0005,
  'BCD': 0.162303,
  'BCH': 123.547963,
  'BCN': 5.3268e-05,
  'BCPT': 0.001504,
  'BEE': 1e-06,
  'BIO': 0

In [37]:
for currency, rate in j_h['rates'].items():
    print(f"{currency} : {rate}")

ABC : 59.99
ACP : 0.014931
ACT : 0.00107
ACT* : 0.017178
ADA : 0.403125
ADCN : 0.00013
ADL : 0.01515
ADX : 0.18734
ADZ : 0.0023
AE : 0.069169
AGI : 0.4267
AIB : 0.005626
AIDOC : 1.627597959e-05
AION : 0.001526
AIR : 0.00976
ALT : 0.565615
AMB : 0.014214
AMM : 0.003387
ANT : 3.67465
APC : 0.0017
APPC : 0.002
ARC : 0.0169
ARCT : 0.00061
ARDR : 0.087504
ARK : 0.81952
ARN : 0.001283
ASAFE2 : 0.4
AST : 0.1173
ATB : 0.017
ATM : 3.43895
AURS : 0.352867
AVAX : 18.36
AVT : 1.14
BAR : 4.215
BASH : 0.0056
BAT : 0.25644
BAY : 0.0644
BBP : 0.0005
BCD : 0.162303
BCH : 123.547963
BCN : 5.3268e-05
BCPT : 0.001504
BEE : 1e-06
BIO : 0.0008
BLC : 0.072132
BLOCK : 0.18
BLU : 0.00054
BLZ : 0.07946
BMC : 0.001922
BNB : 318.409853
BNT : 0.519706
BOST : 0.048
BQ : 7.775e-05
BQX : 2.720931
BRD : 0.007899
BRIT : 0.03
BT1 : 0
BT2 : 0
BTC : 28358.40736
BTCA : 0.00036
BTCS : 0.01201
BTCZ : 0.00015
BTG : 15.349905
BTLC : 9
BTM : 0.078282
BTM* : 0.122609
BTQ : 0.01
BTS : 0.01051
BTX : 0.27027
BURST : 0.017348
CALC :

In [128]:
#data in a dataframe
df8 = pd.DataFrame.from_dict(j_h)
df8


success                        terms                        privacy  \
611      True  https://coinlayer.com/terms  https://coinlayer.com/privacy   
ABC      True  https://coinlayer.com/terms  https://coinlayer.com/privacy   
ACP      True  https://coinlayer.com/terms  https://coinlayer.com/privacy   
ACT      True  https://coinlayer.com/terms  https://coinlayer.com/privacy   
ACT*     True  https://coinlayer.com/terms  https://coinlayer.com/privacy   
...       ...                          ...                            ...   
ZEN      True  https://coinlayer.com/terms  https://coinlayer.com/privacy   
ZIL      True  https://coinlayer.com/terms  https://coinlayer.com/privacy   
ZNY      True  https://coinlayer.com/terms  https://coinlayer.com/privacy   
ZRX      True  https://coinlayer.com/terms  https://coinlayer.com/privacy   
ZSC      True  https://coinlayer.com/terms  https://coinlayer.com/privacy   

       timestamp target  historical        date      rates  
611   1682035146    USD        True  2023-04-20   0.389165  
ABC   1682035146    USD        True  2023-04-20  59.990000  
ACP   1682035146    USD        True  2023-04-20   0.014931  
ACT   1682035146    USD        True  2023-04-20   0.001070  
ACT*  1682035146    USD        True  2023-04-20   0.017178  
...          ...    ...         ...         ...        ...  
ZEN   1682035146    USD        True  2023-04-20   9.877605  
ZIL   1682035146    USD        True  2023-04-20   0.030230  
ZNY   1682035146    USD        True  2023-04-20   0.020000  
ZRX   1682035146    USD        True  2023-04-20   0.266965  
ZSC   1682035146    USD        True  2023-04-20   0.000066  

[387 rows x 8 columns]

Here I converted the data for 2023 into a csv file

In [129]:
#name of file
csv_filename = '2023.csv'

#converting df to csv
df.to_csv(csv_filename, index_label='Currency')

print(f'DataFrame has been saved as "{csv_filename}"')

DataFrame has been saved as "2023.csv"


NOTE: i added the name for the index column as Currency in the csv files, just to make it easier for analysis